### PR TITLE
Translating API tests to Python — part 2

### PR DESCRIPTION
**Pull request by yoni206**
This PR translates the remaining cpp API tests to python.

### CODE REVIEW

File: test/api/python/sep_log_api.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    # assert it
+    slv.assertFormula(y_gt_x)
+
+    # check
+    r = slv.checkSat()
+
+    # If this is UNSAT, we have an issue so bail-out
+    if not r.isSat():
+
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:56]:
```suggestion
```

File: test/api/python/sep_log_api.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        nil_expr = slv.getValueSepNil()
+    except RuntimeError as e:
+        caught_on_nil = True
+
+        # Check we get the correct exception string
+        if str(e) != expected:
+            return -1
+
+    if not caught_on_heap or not caught_on_nil:
+
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:90]:
```suggestion
```

File: test/api/python/sep_log_api.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        "logic theory."
+
+    # test the heap expression
+    try:
+        heap_expr = slv.getValueSepHeap()
+    except RuntimeError as e:
+        caught_on_heap = True
+        # Check we get the correct exception string
+        if str(e) != expected:
+            return -1
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:77]: Use boolean return values for this function and return true when everything behaves as expected.

File: test/api/python/sep_log_api.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    slv.assertFormula(fix_nil)
+
+    # Incremental is disabled due to using separation logic, so don't query
+    # twice!
+
+    r = (slv.checkSat())
+
+    # If this is UNSAT, we have an issue so bail-out
+    if not r.isSat():
+        return -1
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:155]: Ditto

File: test/api/python/sep_log_api.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+    # All tests pass!
+    return 0
+
+
+# check that we get an exception when we should
+check_exception = validate_exception()
+
+if check_exception:
+    assert False
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:234]:
```suggestion
assert validate_exception()
```
Notices my suggestion for changes for `validate_exception()`

File: test/api/python/sep_log_api.py [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+# check the getters
+check_getters = validate_getters()
+
+if check_getters:
+    assert False
```
Comment (aniemetz) [inline test/api/python/sep_log_api.py:240]: Ditto

File: test/api/python/two_solvers.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+# A simple test of multiple SmtEngines.
+##
+
+import pycvc5
+
+s1 = pycvc5.Solver()
+s2 = pycvc5.Solver()
+r = s1.checkEntailed(s1.mkBoolean(True))
+r2 = s2.checkEntailed(s2.mkBoolean(True))
+print(0 if r.isEntailed() and r2.isEntailed() else 1)
```
Comment (aniemetz) [inline test/api/python/two_solvers.py:22]: Since this is a test, we shouldn't just simply print stuff but test for what we excpect.

Comment (yoni206) [reply test/api/python/two_solvers.py]: Thanks, fixed here: 9cd1ca1

**Review by aniemetz — APPROVED** [review]

**Pull request merged**